src/Microsoft.AspNetCore.Http.Connections/HttpConnectionDispatcher.cs

```diff
@@ -419,7 +420,16 @@ private static byte[] GetNegotiatePayload(string connectionId, HttpContext conte
             var ms = new MemoryStream();
             NegotiateProtocol.WriteResponse(response, ms);
 
-            return ms.ToArray();
+            var writer = MemoryBufferWriter.Get();
+            try
+            {
+                NegotiateProtocol.WriteResponse(response, writer);
+                return writer.ToArray();
```

Comment:
I think this `ToArray` is unnessessary. Could pass `MemoryBufferWriter` into this method, serialize response, and then get a `ReadOnlySequence<byte>` from writer and pass that to `context.Response.Body.WriteAsync`.

Just need to add support to `MemoryBufferWriter` for getting a sequence from it.

Reply:
Will look into this tomorrow

Reply:
I have some concerns about this and the thread static caching. I don't think it'll play well together.

Reply:
Another option would be to not use the thread static cache here but expose the `ReadOnlySequence<byte>` and write it directly response.